x/staking/keeper/delegation.go

```diff
@@ -891,55 +973,130 @@ func (k Keeper) CompleteUnbonding(ctx sdk.Context, delAddr sdk.AccAddress, valAd
 	return balances, nil
 }
 
+func (k Keeper) PutUnbondingOpOnHold(ctx sdk.Context, id uint64) (found bool) {
+	_, found = k.GetUnbondingDelegationByUnbondingOpId(ctx, id)
+	if found {
+		return k.PutUnbondingDelegationEntryOnHold(ctx, id)
+	}
+
+	_, found = k.GetRedelegationByUnbondingOpId(ctx, id)
+	if found {
+		return k.PutRedelegationEntryOnHold(ctx, id)
+	}
+
+	// If an entry was not found
+	return false
+}
+
+// This can be called to complete the unbonding of an unbonding operation (unbonding delegation entry or
+// redelegation) that was previously stopped by the BeforeUnbondingDelegationEntryComplete hook in CompleteUnbonding
+func (k Keeper) UnbondingOpCanComplete(ctx sdk.Context, id uint64) (found bool, err error) {
+	_, found = k.GetUnbondingDelegationByUnbondingOpId(ctx, id)
```

Comment:
Same as `PutUnbondingOpOnHold`, first look up the index key and then call the "*CanComplete(indexKey)" functions. 

Also, delete the `indexKey` afterwards.